sympy/stats/tests/test_discrete_rv.py

```diff
@@ -91,6 +91,51 @@ def test_GeometricDistribution():
     Y = Geometric('Y', Rational(3, 10))
     assert coskewness(X, X + Y, X + 2*Y).simplify() == sqrt(230)*Rational(81, 1150)
 
+    # Test that density returns 0 for values outside the support (issue #20031)
+    # The support for Geometric is positive integers {1, 2, 3, ...}
+    assert density(X)(0) == 0  # zero is not in support
+    assert density(X)(-1) == 0  # negative integers not in support
+    assert density(X)(-5) == 0  # negative integers not in support
+    assert density(X)(S(1)/2) == 0  # non-integer not in support
+    assert density(X)(S(5)/2) == 0  # non-integer not in support
+    # Test that density is correct for values inside the support
+    assert density(X)(1) == Rational(1, 5)
+    assert density(X)(2) == Rational(4, 25)
+    assert density(X)(3) == Rational(16, 125)
+
+
+def test_discrete_distribution_call_vs_pdf():
+    X = Geometric('X', Rational(1, 5))
+    z = Symbol('z')
+
+    call_result = density(X)(z)
+    assert call_result == Piecewise(
+        ((Rational(4, 5))**(z - 1) / 5, (z >= 1) & (z < oo) & Eq(z, floor(z))),
+        (0, True)
+    )
+
+    pdf_result = density(X).pdf(z)
+    assert pdf_result == (Rational(4, 5))**(z - 1) / 5
```

Comment:
Here and in the other tests there is no need to assign the result to a variable just to compare it on the next line. It can just be
```python
    assert density(X).pdf(z) == (Rational(4, 5))**(z - 1) / 5
```